ethpm_types/source.py

```diff
@@ -91,6 +91,33 @@ class Source(BaseModel):
     """
     # TODO: Add `SourceId` type and use instead of `str`
 
+    def __repr__(self) -> str:
+        repr_id = "Source"
+        if self.checksum:
+            repr_id = f"{repr_id} {self.checksum.hash}"
+
+        return f"<{repr_id}>"
+
+    def __getitem__(self, number: Union[int, slice]) -> Union[str, List[str]]:
+        """
+        Get a line or slice of lines from ``content``.
+
+        Args:
+            number (int, slice): The line index.
+        """
+
+        self.content = self.fetch_content()
```

Comment:
Might be better to only do this if `self.content` is set and to raise an error saying to use `fetch_content()` if it is not set.